### PR TITLE
Update prismjs to 1.21.0

### PR DESCRIPTION
**Pull request by chenxsan**

### CODE REVIEW

**Comment by wooorm** [comment]
Thanks. Do you need help with this? Btw, the error about eslint-plugin-prettier does not occur when using npm I believe?

**Comment by chenxsan** [comment]
@wooorm Indeed, It works fine with npm while `Cannot find module 'eslint-plugin-prettier'` with yarn.

**Comment by chenxsan** [comment]
I've two problems now:

1. `TypeError: Cannot read property 'keyword' of undefined`:

image

It might be caused by `jsdoc.js` using `typescript` required after it.

2. I don't know how to generate the fixtures, are they generated manually?

**Comment by chenxsan** [comment]
I've pushed another commit to fix the first issue according to https://github.com/wooorm/refractor/commit/c7f8b5d6f76e44f571fd498da616bd62517762cc, but I found the code could be overwritten if we run `npm test` again.

**Comment by wooorm** [comment]
Code is generated, here is where dependencies are detected with Babel: https://github.com/wooorm/refractor/blob/main/script/languages.js#L123

Fixtures are by hand indeed, see here for a couple of lines you can comment out to regenerate: https://github.com/wooorm/refractor/blob/54d22680ceab1abb5f124c82e92135daf33ff37d/test/index.js#L261

**Comment by chenxsan** [comment]
I found the `languages.js` file uses regular expressions to [get the `deps`](https://github.com/wooorm/refractor/blob/main/script/languages.js#L60), I'm wondering whether we can [read its value](https://github.com/PrismJS/prism/blob/388ad996c4b576205de4d4feda69202bd26c1345/components.json#L556) from [`prismjs/components.json`](https://github.com/PrismJS/prism/blob/388ad996c4b576205de4d4feda69202bd26c1345/components.json#L556) where they have this `require` attribute? I've pushed a new commit here https://github.com/wooorm/refractor/pull/32/commits/89421624632a2ce2cd70df527fda2dbc632adc25. As you can see it changed a lot of code, would love to hear your thought about it.

**Comment by wooorm** [comment]
Oh nice! That as 't there when I originally made refractor. For your commit, you should remove the defaults: https://github.com/wooorm/refractor/blob/main/script/bundled.json

**Comment by chenxsan** [comment]
Both [`script/list.js`](https://github.com/wooorm/refractor/blob/main/script/list.js#L8) and [`script/support.js`](https://github.com/wooorm/refractor/blob/main/script/support.js#L9) depends on `bundled.json`, so I can't remove it naively.

**Comment by thibaut-singlefile** [comment]
Thanks for creating this PR! We're waiting for this to be merged in and a new version to be released to upgrade prismjs.

**Comment by wooorm** [comment]
I’m sorry that my comment was unclear! I meant: you can *use* the bundled file, to remove stuff. For example, JavaScript requires are not needed: https://github.com/wooorm/refractor/pull/32/files#diff-a067b044e29ecf0bf62f582853f9b9d6R2, because they are already registered! I think that’s the main reason for your earlier comment: “As you can see it changed a lot of code, would love to hear your thought about it.”

**Comment by chenxsan** [comment]
@wooorm It's me misunderstood your comment. I've pushed a new commit, hope it's right this time :)

File: package.json [hunk 1]
```diff
@@ -37,7 +37,7 @@
   "dependencies": {
     "hastscript": "^5.0.0",
     "parse-entities": "^2.0.0",
-    "prismjs": "~1.20.0"
+    "prismjs": "^1.21.0"
```
Comment (wooorm) [inline package.json:40]:
```suggestion
    "prismjs": "~1.21.0"
```

The tilde is there for a reason. Prism sometimes breaks in minors, so we use tildes.

File: script/languages.js [hunk 2]
```diff
@@ -57,18 +54,21 @@ function generate(name, callback) {
...
-      findAll(doc, extendRegex),
-      findAll(doc, cloneRegex),
-      findAll(doc, insertRegex)
-    )
-    anyAlias = findAll(doc, anyAliasRegex).join('\n')
-    aliases = findAll(anyAlias, aliasRegex).filter((d) => d !== name)
+    deps =
+      typeof componentsJson.languages[name].require === 'undefined'
+        ? []
+        : componentsJson.languages[name].require
```
Comment (wooorm) [inline script/languages.js:60]:
```suggestion
    deps = componentsJson.languages[name].require || []
```

**Comment by wooorm** [comment]
Thanks! Released in 3.1.0! ✨

**Pull request merged**